liquigraph-core/src/main/java/org/liquigraph/core/io/GraphJdbcConnector.java

```diff
@@ -25,13 +26,21 @@
 
 public class GraphJdbcConnector implements LiquigraphJdbcConnector {
 
+    /**
+     * Acquires a new connection to the configured instance
+     * and tries to lock it (fail-fast).
+     *
+     * @see {@link LockableConnection}
+     * @param configuration Liquigraph settings
+     * @return JDBC connection
+     */
     @Override
     public final Connection connect(Configuration configuration) {
         try {
             Class.forName("org.neo4j.jdbc.Driver");
             Connection connection = DriverManager.getConnection(makeUri(configuration));
             connection.setAutoCommit(false);
-            return connection;
+            return new LockableConnection(connection);
```

Comment:
Actually, the shutdown hook is also leaked. It's hard to manage all that inside of a constructor. It might be better to use a factory method to create the `LockableConnection` with its final fields, then do the locking and properly manage the failures to remove everything set up so far (remove the hook, maybe close the delegate connection since it's also closed by `close()`).